### DISCUSSION TITLE
ShutterButton not working as expected

### DISCUSSION THREAD

**Discussion by gsasha** [arendst/Tasmota#13223] [Q&A] [answered]
I have a Sonoff dual R3 flashed with tasmota, configured for blinds control. Commands from the console seem to be working.

However, I'm unable to set button functions - whatever I do, the buttons only directly control the relays, and I'm unable to bind any functions to single/double clicks etc.

Here's the configuration:
```
20:00:29.474 CMD: status 13
20:00:29.492 MQT: stat/tasmota_2C60C8/STATUS13 = {"StatusSHT":{"SHT0":{"Relay1":1,"Relay2":2,"Open":140,"Close":123,"50perc":55,"Delay":0,"Opt":"0100","Calib":[280,252,812,900,1000],"Mode":"1"}}}
20:00:46.137 CMD: shutterbutton1
20:00:46.155 MQT: stat/tasmota_2C60C8/RESULT = {"ShutterButton1":"1 100 50 74 100 - - - - -|2 0 50 24 0 - - - - -"}
```

What am I missing?

**Comment by TrDA-hab** [comment]
SetOption73 ON // Enable detach buttons from relays.

Assigns button#1 to Switching between motors (1x press up, 2x press down):
Rule1
ON button1#state=10 DO Backlog Power1 1; Power2 0 ENDON // 1x press up to run motor#1.
ON button1#state=11 DO Backlog Power2 1; Power1 0 ENDON // 3x press up to run motor#2.
Rule1 1 // Enable Rule1.

**Comment by gsasha** [comment]
So I've done the following:
```
setoption73 on
shutterbutton1 1 up
shutterbutton1 2 down
```
The buttons still seem to have direct relay control. Even rebooted for a good measure.

 Will it go away if I define the rule? That'd be strange.

In any case, (assuming that it gets working) SetOption73 is not mentioned anywhere on the page, and I guess it should be in the [Button Control](https://tasmota.github.io/docs/Blinds-and-Shutters/#button-control) section.

**Comment by TrDA-hab** [comment]
Run commands in the console for configuring the drive:

SetOption73 1   // Enable detach buttons from relays.

ShutterButton1 1 updown 0  // Assigns button1 to act as an "up and down" button (1x press up, 2x press down) for Shutter#1.

p.s.

Do you have a physical button?
Do you double-click with the mouse?
Or  ... ?

**Reply by gsasha** [reply]
I use a Sonoff Dual R3. It's a device with two relays and two signal inputs. It's connected to the standard blinds switch (i.e., two push buttons that are wired to the signal inputs), and the two line outputs of the sonoff are wired to the up/down motors.

The tasmota UI has up/down buttons and a slider - all of these work as expected, i.e., open and close the blinds fully.

It's the physical buttons that seem to be wired to relays, despite SetOption73

Ah, notice that the controller in the Sonoff is ESP32, so it's Tasmota32 - could it be that there is a bug in this specific version?

**Comment by gsasha** [comment]
Here I made a short video that demonstrates the problem: https://photos.app.goo.gl/g2fqPgwPN4vfBjar8
It shows:
1. if I press the up/down buttons in the UI, it opens/closes automatically till the end
2. if I move the slider to a certain position, it scrolls as intended
3. if I hold the physical buttons (connected to the signal inputs), the blinds go up/down for as long as I hold the button
4. if I press the buttons momentarily, it just moves the blinds a little bit, for as long as the button is pressed, rather than opening/closing the shutter fully as I'd expect "ShutterButton1 1 up" to do.

**Comment by gsasha** [comment]
Ok, think I found it. The GPIOs in the configuration were set up as "switch", not "button". Changed them to be "button", now I see that a button works as expected (i.e., a single click activates "close till the end" mode). Only one of the buttons seems to work now, looks like I managed to disable something, debugging.

**Comment by TrDA-hab** [comment]
Do you understand (?) The difference between
  - "I press buttons for a moment."
  - "and I make two clicks on the button."
  - and there is no (!) function to fully open / close the shutter ... while the button is pressed ... only one or two presses !!!

Try using this:
`SetOption73 1` // Enable detach buttons from relays.
`ShutterButton1 1 updown 0` // Assigns button1 to act as an "up and down" button (1x press up, 2x press down) for Shutter # 1.

And try `one tap` or `double tap` on the physical button.

**p.s. long press is not (!) supported.**

**Comment by gsasha** [comment, marked as answer]
Ok, I managed to solve it. To summarize the steps I needed:
- `SetOption73 1` ~~to detach the buttons~~. UPD: No need for this: the problem is the Button configuration.
- In configuration, made sure that the GPIOs are configured as "Button" (they were "Switch" by default). UPD: SwitchMode can be used instead.
- `Shutterbutton1 3 up` (something I don't understand got borked in my configuration: "button1" seems to be ignored, so I mapped the GPIO as "button 3" instead.
- `Shutterbutton1 2 down`

I think it's important to document in the "Shutters and Blinds" section that the GPIOs must be configured as "Button", not "Switch".

Now tapping the physical buttons indeed activates the "full close/full open" action, and a second tap stops the movement.

It would be nice to configure a timeout, say a second, after which a held button would switch to "direct" mode, i.e., the motor moves as long as the button is held. I guess the software doesn't have that implemented - do you think it's a good idea to file a feature request for that?

**Reply by crllsn** [reply]
I'm a newbie here but I think you should not mess gpios but rather use SWITCHMODE command. In your case, I think SWITCHMODE 3 is the proper one.

**Reply by gsasha** [reply]
Thanks, seems you are right. I won't touch my config already, but this is a better approach.
And SetOption73 is actually not necessary, it's the switchmode that was wrong.

**Comment by gsasha** [comment]
Thanks for your help!!!
